server/appartners_server.py

```diff
@@ -95,5 +118,70 @@ def login():
         return jsonify({"errorMessage": str(e)}), 500
 
 
+@app.route('/create-profile', methods=['POST', 'OPTIONS'])
+def create_profile():
+    try:
+        if request.method == 'OPTIONS':
+            return {}, 200
+        profile_data = request.get_json()
+        user_id = profile_data.get('user_id')
+        profile_bio = profile_data.get('profile_bio')
+        photo_url = profile_data.get('photo_url')
+        first_name = profile_data.get('first_name')
+        last_name = profile_data.get('last_name')
+        sex = profile_data.get('sex')
+        birthday = profile_data.get('birthday')
+        age = profile_data.get('age')
+        smoking = profile_data.get('smoking')
+        like_animals = profile_data.get('like_animals')
+        keeps_kosher = profile_data.get('keeps_kosher')
+        first_roomate_appartment = profile_data.get('first_roomate_appartment')
+        profession = profile_data.get('profession')
+        status = profile_data.get('status')
+        hobbies = profile_data.get('hobbies')
+        has_animals = profile_data.get('has_animals')
+        alergies = profile_data.get('alergies')
+
+        if not first_name or not first_name.strip():
+            logger.log_error(f"Profile creation falied for user - {user_id}. First name is mandatory value.")
+            return jsonify({"errorMessage": "First name is required"}), 400
+        if not last_name or not last_name.strip():
+            logger.log_error(f"Profile creation falied for user - {user_id}. Last name is mandatory value.")
+            return jsonify({"errorMessage": "Last name is required"}), 400
+        if not sex or not sex.strip():
+            logger.log_error(f"Profile creation falied for user - {user_id}. Gender is mandatory value.")
+            return jsonify({"errorMessage": "Sex is required"}), 400
+        if not birthday:
+            logger.log_error(f"Profile creation falied for user - {user_id}. Birthday is mandatory value.")
+            return jsonify({"errorMessage": "Birthday is required"}), 400
+        if not validate_date_format(birthday):

```

Comment:
probably will be deleted. we will need to give an option to insert only a valid date

Reply:
need to add a function to convert dd/mm//yyyy to yyyy-mm-dd